backend/http_backend/http_backend.go

```diff
@@ -0,0 +1,107 @@
+package httpbackend
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/skroutz/downloader/job"
+)
+
+// DefaultClientTimeoutSec defines a default timeout in seconds for our http client
+const DefaultClientTimeoutSec = 30
+
+var (
+	// Based on http.DefaultTransport
+	//
+	// See https://golang.org/pkg/net/http/#RoundTripper
+	transport = &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   10 * time.Second, // was 30 * time.Second
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
```

Comment:
I'd suggest those settings (or at least the most important subset of them) be configurable via our config file similar to kafka-specific settings.

Reply:
@dblia We won't have to mess with these often (if at all), so I'd suggest to leave them as is, at least for now.